src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -1526,7 +1526,10 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
         }
 
       if (deploy_has_bool_option (self, "reboot"))
-        rpmostreed_reboot (cancellable, error);
+        {
+          sd_journal_print (LOG_INFO, "daemon-initiated reboot is deprecated");
```

Comment:
I was just wondering why we don't log a warning in the other methods too (e.g. `initramfs_etc_transaction_execute()`, `kernel_arg_transaction_execute()`, `rollback_transaction_execute()`, etc.)? Or if they should never be used anyways, since no one should be calling with that option, what if we just removed `rpmostreed_reboot()` from those functions?

For Zincati's purposes though, since it cares about the `rpm-ostree finalize-deployment` CLI command, which eventually calls the `finalize_deployment_transaction_execute()` daemon function, it looks like the reboot will still be on the daemon side. Is it possible to move the reboot to the client side for `rpm-ostree finalize-deployment` too?

Reply:
> initramfs_etc_transaction_execute()

Good catch!  I missed at least that one because it used a different function for looking up the `"reboot"` flag so my code search missed it.  Will update.

> Is it possible to move the reboot to the client side for rpm-ostree finalize-deployment too?

Ah right, I should have mentioned this in the PR.  I looked at `finalize-deployment` but the problem I see is that (this relates to your first point too) the DBus API should generally be considered a stable interface.

For example, https://github.com/cockpit-project/cockpit-ostree/blob/master/src/client.js and https://gitlab.gnome.org/GNOME/gnome-software/-/blob/master/plugins/rpm-ostree/gs-plugin-rpm-ostree.c 

So what we'd likely need to do for e.g. `finalize-deployment` is *add* a `reboot` option, make it the default, and have our CLI pass `FALSE` or so.

Probably cleanest as a followup PR?

Reply:
(Decided to fix `FinalizeDeployment` in this PR too, then we can log the deprecation in one place)

Reply:
I think CI highlighted a regression here. The previous reboot logic was reachable by non-privileged processes through polkit ACL. This new code instead moves it client-side, where the same processes cannot directly trigger a reboot. That is the case for `zincati`:
```
zincati[1029]: [ERROR] failed to finalize deployment: rpm-ostree finalize-deployment failed:
zincati[1029]:     Failed to set wall message, ignoring: Interactive authentication required.
zincati[1029]:     Failed to reboot system via logind: Interactive authentication required.
zincati[1029]:     Failed to open initctl fifo: Permission denied
zincati[1029]:     Failed to talk to init daemon.
```

Reply:
Mmm.  OK what about actually pushing the `reboot` all the way into zincati, i.e. we extend the polkit rules there to allow it?

Reply:
That'd be fine for zincati specifically, I'll look into if/how it can be done directly in polkit (I think so) or capabilities.
I am not sure about consumers though, in particular the ones running containerized under their own PID namespace where "reboot" has a bit more nuances.